FWCore/ParameterSet/src/ConfigurationDescriptions.cc

```diff
@@ -147,6 +147,54 @@ namespace edm {
                       std::cref(baseType_),
                       std::cref(pluginName_),
                       std::ref(usedCfiFileNames)));
+    if (defaultDescDefined_) {
+      writeClassFile(defaultDesc_);
+    } else if (descriptions_.size() == 1) {
+      writeClassFile(descriptions_.begin()->second);
+    }
+  }
+
+  void ConfigurationDescriptions::writeClassFile(ParameterSetDescription const& iDesc) const {
+    std::string pluginName = pluginName_;
+    auto found = pluginName.find("::");
+    while (found != std::string::npos) {
+      pluginName.replace(found, 2, "_");
+      found = pluginName.find("::");
+    }
+    //Symbols that can appear in our plugin names but can't in python function names
+    const std::string toReplace("@<>,");
+    found = pluginName.find_first_of(toReplace);
+    while (found != std::string::npos) {
+      pluginName.replace(found, 1, "_");
+      found = pluginName.find_first_of(toReplace, found);
+    }
```

Comment:
IIUC this means that a plugin with a name `FooProducer@alpaka` would lead to a generating function with the name `FooProducer_alpaka`. We may want to think about this more in the future (e.g. whether to drop the `_alpaka` part).

Reply:
That is how the name would be modified.